sw/otbn/crypto/p384_curve_point_valid.s

```diff
@@ -0,0 +1,204 @@
+/* Copyright lowRISC contributors. */
+/* Licensed under the Apache License, Version 2.0, see LICENSE for details. */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+/**
+ * Curve point validation for curve P-384.
+ *
+ * This binary has the following modes of operation:
+ * 1. MODE_SHARED_KEYGEN: check if a given public key for ECDH shared key
+ *                        generation is a valid curve point
+ */
+
+ /**
+ * Mode magic values generated with
+ * $ ./util/design/sparse-fsm-encode.py -d 6 -m 4 -n 11 \
+ *    --avoid-zero -s 3660400884
+ *
+ * Call the same utility with the same arguments and a higher -m to generate
+ * additional value(s) without changing the others or sacrificing mutual HD.
+ *
+ * TODO(#17727): in some places the OTBN assembler support for .equ directives
+ * is lacking, so they cannot be used in bignum instructions or pseudo-ops such
+ * as `li`. If support is added, we could use 32-bit values here instead of
+ * 11-bit.
+ */
+.equ MODE_SHARED_KEY, 0x5ec
+
+.section .text.start
+start:
+  /* Init all-zero register. */
+  bn.xor    w31, w31, w31
+
+  /* Read the mode and tail-call the requested operation. */
+  la        x2, mode
+  lw        x2, 0(x2)
+
+  addi      x3, x0, MODE_SHARED_KEY
+  beq       x2, x3, p384_curve_point_valid
```

Comment:
Since there's only one option, I think you can drop the mode parameter here.

Reply:
Makes sense!